kotlin/README.md

```diff
@@ -0,0 +1,616 @@
+corresponding declaration:
+
+```
+@Test fun foo() { /*...*/ }
+```
+
+### Function formatting:
+If the function signature doesn't fit on a single line, use the following syntax:
+
+```
+fun longMethodName( 
+    argument: ArgumentType = defaultValue, 
+    argument2: AnotherArgumentType 
+): ReturnType { 
+    // body 
+}
+```
+
+Use regular indent (4 spaces) for function parameters.
+
+Prefer using an expression body for functions with the body consisting of a single 
+expression.
+
+```
+// bad 
+fun foo(): Int { 
+    return 1 
+} 
+
+// good
+fun foo() = 1
+```
+
+### Expression body formatting:
+If the function has an expression body that doesn't fit in the same line as the 
+declaration, put the = sign on the first line. Indent the expression body by 4 
+spaces.
+
+```
+fun f(x: String) = 
+    x.length
+```
+
+### Property formatting:
+For very simple read-only properties, consider one-line formatting: 
+
+```
+val isEmpty: Boolean get() = size == 0
+```
+
+For more complex properties, always put get and set keywords on separate lines:
+
+```
+val foo: String 
+get() { /*...*/ }
+```
+
+For properties with an initializer, if the initializer is long, add a line break 
+after the equals sign and indent the initializer by four spaces:
+
+```
+private val defaultCharset: Charset? = 
+    EncodingRegistry.getInstance().getDefaultCharsetForPropertiesFiles(file)
+```
+
+### Formatting control flow statements:
+If the condition of an if or when the statement is multiline, always use curly 
+braces around the body of the statement. Indent each subsequent line of the 
+condition by 4 spaces relative to statement begin. Put the closing parentheses 
+of the condition together with the opening curly brace on a separate line:
+
+```
+if (!component.isSyncing && 
+    !hasAnyKotlinRuntimeInScope(module) 
+) { 
+    return createKotlinNotConfiguredPanel(module) 
+}
+```
+
+Put the else , catch , finally keywords, as well as the while keyword of a do/while 
+loop, on the same line as the preceding curly brace:
+
+```
+if (condition) {
+    // body 
+} else {
+    // else part 
+} 
+
+
+try { 
+    // body 
+} finally { 
+    // cleanup 
+}
+```
+
+In a when statement, if a branch is more than a single line, consider separating it
+from adjacent case blocks with a blank line:
+
+```
+private fun parsePropertyValue(propName: String, token: Token) {
+	  when (token) {
+		    is Token.ValueToken ->
+			  callback.visitValue(propName, token.value)
+		
+		    Token.LBRACE -> { // …
+        }
+    }
+}
+```
+
+Put short branches on the same line as the condition, without braces.
+
+```
+when (foo) {
+    true -> bar() // good 
+    false -> { baz() } // bad 
+}
+```
+
+### Method call formatting:
+In long argument lists, put a line break after the opening parenthesis. Indent 
+arguments by 4 spaces. Group multiple closely related arguments on the same line.
+
+```
+drawSquare( 
+    x = 10, y = 10, 
+    width = 100, height = 100, 
+    fill = true )
```

Comment:
Doesn't it look better with the closing parenthesis in a new line? If not, I'd remove the space before it.

Reply:
I think it should be on the next line